backend/app/services/langgraph_chat/tools/skills/file_reader.py

```
@@ -0,0 +1,239 @@
+        Raises:
+            ValueError: If path is outside workspace root
+        """
+        # Remove leading slash if present
+        file_path = file_path.lstrip("/")
+
+        # Construct full path
+        full_path = os.path.abspath(os.path.join(self.workspace_root, file_path))
+
+        # Security check: ensure path is within workspace
+        if not full_path.startswith(os.path.abspath(self.workspace_root)):
+            raise ValueError(f"Path traversal detected: {file_path}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)